### PR TITLE
feat: assert actual result and expected result are equal in code

### PR DESCRIPTION
**Pull request by ypinhsuan**
Arrays are printed differently for each language. E.g. `[ 1, 2 ]` `[1,2]`, `[1, 2]`
So we will compare the actual and expected output in the submitted code

Judge0 will return `True` or `False` in stdout

Part of #207

### CODE REVIEW

File: backend/src/judge/middleware/java.ts [hunk 1]
```diff
@@ -56,22 +56,32 @@ export class JavaMiddleware extends JudgeMiddleware {
...
+    let isEqual = "";
+    if (codePrototype.returnType.includes("[]")) {
+      const output = this.output.replace(/^\[/, "{").replace(/\]$/, "}");
+      expectedOutput = `${codePrototype.returnType} expectedOutput = ${output};`;
+      isEqual = `boolean isEqual = Arrays.equals(res, expectedOutput);`;
+    } else {
+      expectedOutput = `${codePrototype.returnType} expectedOutput = ${this.output};`;
+      isEqual =
+        "boolean isEqual = res.toString().equals(expectedOuput.toString());";
+    }
```
Comment (dextertanyj) [inline backend/src/judge/middleware/java.ts:69]: Actually can we just stringify every time?

File: backend/src/judge/middleware/javascript.ts [hunk 2]
```diff
@@ -39,6 +39,14 @@ export class JavascriptMiddleware extends JudgeMiddleware {
       );
     }
 
+    let isEqual = "";
+    if (codePrototype.returnType.includes("[]")) {
+      isEqual =
+        "const isEqual = JSON.stringify(result) == JSON.stringify(expectedOutput);";
+    } else {
+      isEqual = "const isEqual = result == expectedOutput;";
+    }
```
Comment (dextertanyj) [inline backend/src/judge/middleware/javascript.ts:48]: Same as above

File: backend/src/judge/middleware/cpp.ts [hunk 3]
```diff
@@ -66,33 +70,29 @@ export class CppMiddleware extends JudgeMiddleware {
...
-        `    if (i != 0) {\n` +
-        `      ss << ", ";\n` +
-        `    }\n` +
-        `    ss << res[i];\n` +
-        `  }\n` +
-        `  ss << "]";\n` +
-        `  std::cout << ss.str() << std::endl;\n`;
+      const output = this.output.replace(/^\[/, "{").replace(/\]$/, "}");
+      expectedOutput = `${codePrototype.returnType} expectedOutput${output};`;
+      isEqual = `bool isEqual = std::equal(res.begin(), res.end(), expectedOutput.begin());`;
```
Comment (dextertanyj) [inline backend/src/judge/middleware/cpp.ts:78]: Do we need to check that `res` is the same size as `expectedOutput`?

File: backend/src/judge/middleware/java.ts (hunk 1 above)
Comment (ypinhsuan) [reply backend/src/judge/middleware/java.ts:69]: Arrays in java need to be handled separately as `arr.toString()` returns hexadecimal of the hashcode which is different for arrays that are equal.

File: backend/src/judge/middleware/javascript.ts (hunk 2 above)
Comment (ypinhsuan) [reply backend/src/judge/middleware/javascript.ts]: Fixed

File: backend/src/judge/middleware/cpp.ts (hunk 3 above)
Comment (ypinhsuan) [reply backend/src/judge/middleware/cpp.ts]: Yep I forgot about this. Thanks!

**Review by dextertanyj — APPROVED** [review]
lgtm

**Pull request merged**